File: Exemplos SmartPOS/App_IntentDigitalHUB_Ionic/Elgin_IntentDigitalHUB_SmartPOS_Ionic-font/src/intentDigitalHubService/ELGINPAY/Commands/SetPersonalizacao.ts
import {ElginPayCommand} from '../ElginPayCommand';

export class SetPersonalizacao extends ElginPayCommand {
  readonly iconeToolbar: string;
  readonly fonte: string;
  readonly corFonte: string;
  readonly corFonteTeclado: string;
  readonly corFundoToolbar: string;
  readonly corFundoTela: string;
  readonly corTeclaLiberadaTeclado: string;
  readonly corFundoTeclado: string;
  readonly corTextoCaixaEdicao: string;
  readonly corSeparadorMenu: string;

  constructor(
    iconeToolbar: string,
    fonte: string,
    corFonte: string,
    corFonteTeclado: string,
    corFundoToolbar: string,
    corFundoTela: string,
    corTeclaLiberadaTeclado: string,
    corFundoTeclado: string,
    corTextoCaixaEdicao: string,
    corSeparadorMenu: string,
  ) {
    super('setPersonalizacao');
    this.iconeToolbar = iconeToolbar;
    this.fonte = fonte;
    this.corFonte = corFonte;
    this.corFonteTeclado = corFonteTeclado;
    this.corFundoToolbar = corFundoToolbar;
    this.corFundoTela = corFundoTela;
    this.corTeclaLiberadaTeclado = corTeclaLiberadaTeclado;
    this.corFundoTeclado = corFundoTeclado;
    this.corTextoCaixaEdicao = corTextoCaixaEdicao;
    this.corSeparadorMenu = corSeparadorMenu;
  }

  functionParameters(): object {
    return {
      iconeToolbar: this.iconeToolbar,
      fonte: this.fonte,
      corFonte: this.corFonte,
      corFonteTeclado: this.corFonteTeclado,
      corFundoToolbar: this.corFundoToolbar,
      corFundoTela: this.corFundoTela,
      corTeclaLiberadaTeclado: this.corTeclaLiberadaTeclado,
      corFundoTeclado: this.corFundoTeclado,
      corTextoCaixaEdicao: this.corTextoCaixaEdicao,
      corSeparadorMenu: this.corSeparadorMenu,
    };
  }
}

File: Exemplos SmartPOS/App_IntentDigitalHUB_Kotlin/Elgin_IntentDigitalHUB_SmartPOS_Kotlin-font/app/src/main/java/com/elgin/kotlin_intent_digital_hub_smartpos/IntentDigitalHubService/ELGINPAY/Commands/SetPersonalizacao.kt
package com.elgin.kotlin_intent_digital_hub_smartpos.IntentDigitalHubService.ELGINPAY.Commands

import com.elgin.kotlin_intent_digital_hub_smartpos.IntentDigitalHubService.ELGINPAY.ElginPayCommand
import com.elgin.kotlin_intent_digital_hub_smartpos.IntentDigitalHubService.IntentDigitalHubModule
import com.google.gson.JsonObject


class SetPersonalizacao(
    private val iconeToolbar: String?,
    private val fonte: String?,
    private val corFonte: String?,
    private val corFonteTeclado: String?,
    private val corFundoToolbar: String?,
    private val corFundoTela: String?,
    private val corTeclaLiberadaTeclado: String?,
    private val corFundoTeclado: String?,
    private val corTextoCaixaEdicao: String?,
    private val corSeparadorMenu: String?
): ElginPayCommand("setPersonalizacao") {


    override fun functionParametersJson(): JsonObject? {
        val functionParametersJson = JsonObject()
        functionParametersJson.addProperty("iconeToolbar", iconeToolbar)
        functionParametersJson.addProperty("fonte", fonte)
        functionParametersJson.addProperty("corFonte", corFonte)
        functionParametersJson.addProperty("corFonteTeclado", corFonteTeclado)
        functionParametersJson.addProperty("corFundoToolbar", corFundoToolbar)
        functionParametersJson.addProperty("corFundoTela", corFundoTela)
        functionParametersJson.addProperty("corTeclaLiberadaTeclado", corTeclaLiberadaTeclado)
        functionParametersJson.addProperty("corFundoTeclado", corFundoTeclado)
        functionParametersJson.addProperty("corTextoCaixaEdicao", corTextoCaixaEdicao)
        functionParametersJson.addProperty("corSeparadorMenu", corSeparadorMenu)
        return functionParametersJson
    }
}